api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -543,20 +560,8 @@ message HttpConnectionManager {
   // emitted by the connection manager.
   repeated config.accesslog.v3.AccessLog access_log = 13;
 
-  // The interval to flush the above access logs. By default, the HCM will flush exactly one access log
-  // on stream close, when the HTTP request is complete. If this field is set, the HCM will flush access
-  // logs periodically at the specified interval. This is especially useful in the case of long-lived
-  // requests, such as CONNECT and Websockets. Final access logs can be detected via the
-  // `requestComplete()` method of `StreamInfo` in access log filters, or thru the `%DURATION%` substitution
-  // string.
-  // The interval must be at least 1 millisecond.
-  google.protobuf.Duration access_log_flush_interval = 54
-      [(validate.rules).duration = {gte {nanos: 1000000}}];
-
-  // If set to true, HCM will flush an access log once when a new HTTP request is received, after request
-  // headers have been evaluated, before iterating through the HTTP filter chain.
-  // Details related to upstream cluster, such as upstream host, will not be available for this log.
-  bool flush_access_log_on_new_request = 55;
+  // Additional access log options for HTTP connection manager.
+  HcmAccessLogOptions access_log_options = 54;
```

Comment:
As this field exists in main branch for more than 14 days, we cannot remove a field per API versioning policy: https://github.com/envoyproxy/envoy/blob/main/api/API_VERSIONING.md

Reply:
@lizan  Since this is still in current.yaml, is it reasonable to deprecate before it goes out to a new minor release? I'm finding myself wanting to add at least one more logging option.

Reply:
I took the deprecation path and adjusted the PR. can you please review again @lizan?